src/node_crypto.cc

```diff
@@ -4062,10 +4063,33 @@ void ECDH::GenerateKeys(const FunctionCallbackInfo<Value>& args) {
 
   if (!EC_KEY_generate_key(ecdh->key_))
     return env->ThrowError("Failed to generate EC_KEY");
-
-  ecdh->generated_ = true;
```

Comment:
I mean, perhaps it should be set from `GeneratePublicKey` too? Or be a flag field with two possible flags: `GeneratedPrivate` and `GeneratedPublic`?